services/api-server/src/simcore_service_api_server/modules/director_v2.py

```diff
@@ -1,48 +1,180 @@
+from httpx import HTTPStatusError, codes
+from models_library.projects_pipeline import ComputationTask
+from models_library.projects_state import RunningState
+from pydantic import AnyHttpUrl, Field, PositiveInt
+from starlette import status
 
 from ..core.settings import DirectorV2Settings
-from ..utils.client_base import BaseServiceClientApi
-from ..utils.client_decorators import JsonDataType, handle_errors, handle_retry
+from ..models.schemas.jobs import PercentageInt
+from ..utils.client_base import BaseServiceClientApi, setup_client_instance
 
 logger = logging.getLogger(__name__)
 
-# Module's setup logic ---------------------------------------------
 
+# API MODELS ---------------------------------------------
+# NOTE: as services/director-v2/src/simcore_service_director_v2/models/schemas/comp_tasks.py
+# TODO: shall schemas of internal APIs be in models_library as well?? or is against
 
-def setup(app: FastAPI, settings: DirectorV2Settings) -> None:
-    if not settings:
-        settings = DirectorV2Settings()
-
-    def on_startup() -> None:
-        logger.debug("Setup %s at %s...", __name__, settings.base_url)
-        DirectorV2Api.create(
-            app,
-            client=httpx.AsyncClient(base_url=settings.base_url),
-            service_name="director_v2",
-        )
 
-    async def on_shutdown() -> None:
-        client = DirectorV2Api.get_instance(app)
-        if client:
-            await client.aclose()
-        logger.debug("%s client closed successfully", __name__)
+class ComputationTaskOut(ComputationTask):
+    url: AnyHttpUrl = Field(
+        ..., description="the link where to get the status of the task"
+    )
+    stop_url: Optional[AnyHttpUrl] = Field(
+        None, description="the link where to stop the task"
+    )
 
-    app.add_event_handler("startup", on_startup)
-    app.add_event_handler("shutdown", on_shutdown)
+    def guess_progress(self) -> PercentageInt:
+        # guess progress based on self.state
+        # FIXME: incomplete!
+        if self.state in [RunningState.SUCCESS, RunningState.FAILED]:
+            return 100
+        return 0
 
 
 # API CLASS ---------------------------------------------
 
 
+@contextmanager
+def handle_errors_context(project_id: UUID):
+    try:
+
+        yield
+
+    # except ValidationError
+    except HTTPStatusError as err:
+        msg = (
+            f"Failed {err.request.url} with status={err.response.status_code}: {err.response.json()}",
+        )
+        if codes.is_client_error(err.response.status_code):
+            # client errors are mapped
+            logger.debug(msg)
+            if err.response.status_code == status.HTTP_404_NOT_FOUND:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail=f"Job {project_id} not found",
+                ) from err
+
+            raise err
+
+        # server errors are logged and re-raised as 503
```

Comment:
It is also possible to get something different here, maybe 401 and 500? Maybe the message is misleading?

Reply:
yes, error handling is not complete. I mention it in the comments as something I will do separately in next PR